mlx/coverity_services.py

```diff
@@ -1,253 +1,203 @@
-
-        # use stream name as an initial glob pattern
-        filter_spec.namePattern = stream_name
-
-        # get all the streams that match
-        streams = self.client.service.getStreams(filter_spec)
-
-        # find the one with a matching name
-        for stream in streams:
-            if compare_strings(stream.id.name, stream_name):
-                return stream
-        return None
-
-    # get a list of the snapshots in a named stream
-    def get_snapshot_for_stream(self, stream_name):
-        '''Get snapshot object for given stream name'''
-        stream_id = self.client.factory.create('streamIdDataObj')
-        stream_id.name = stream_name
-        # optional filter specification
-        filter_spec = self.client.factory.create('snapshotFilterSpecDataObj')
-        # return a list of snapshotDataObj
-        return self.client.service.getSnapshotsForStream(stream_id, filter_spec)
+        Args:
+            stream (str): The stream name
+        """
+        url = self.base_url.rstrip('/') + f"/streams/{stream}"
+        self._request(url)
 
-    @staticmethod
-    def get_snapshot_id(snapshots, idx=1):
-        '''Get the nth snapshot (base 1) - minus numbers to count from the end backwards (-1 = last)'''
-        if bool(idx):
-            num_snapshots = len(snapshots)
-            if idx < 0:
-                required = num_snapshots + idx + 1
-            else:
-                required = idx
-
-            if abs(required) > 0 and abs(required) <= num_snapshots:
-                # base zero
-                return snapshots[required - 1].id
-        return 0
-
-    def get_snapshot_detail(self, snapshot_id):
-        '''Get detailed information about a single snapshot'''
-        snapshot = self.client.factory.create('snapshotIdDataObj')
-        snapshot.id = snapshot_id
-        # return a snapshotInfoDataObj
-        return self.client.service.getSnapshotInformation(snapshot)
-
-    def get_checkers(self):
-        '''Get a list of checkers from the service'''
+    def retrieve_issues(self, filters):
+        """Retrieve issues from the server (Coverity Connect).
+
+        Args:
+            filters (json): The filters as json
+
+        Returns:
+            dict: The response
+        """
+        url = self.base_url.rstrip('/') + \
+            "/issues/search?includeColumnLabels=true&offset=0&queryType=bySnapshot&rowCount=-1&sortOrder=asc"
+        return self._request(url, filters)
+
+    def retrieve_column_keys(self):
+        """Retrieves the column keys and associated display names.
+
+        Returns:
+            list[dict]: A list of dictionaries where the keys of each dictionary are 'columnKey' and 'name'
+        """
+        if not self._columns:
+            url = f"{self.base_url.rstrip('/')}/issues/columns?queryType=bySnapshot&retrieveGroupByColumns=false"
+            self._columns = self._request(url)
+        return self.columns
```

Comment:
I think that returning a list of namedtuple(name, column_key) would be more user-friendly.

Reply:
Resolved in aab78642dbefa9a8ec875efc7ee6dc8e9f555400